tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/FilterStaticFrameworksTaskTest.cs

```diff
@@ -0,0 +1,225 @@
+			var customExecutablePath = Path.Combine (frameworkDir, executableName);
+			File.WriteAllBytes (customExecutablePath, CreateMinimalMachODylib ());
+
+			// Act: Create and execute the task
+			var task = CreateTask<FilterStaticFrameworks> ();
+			task.FrameworkToPublish = new ITaskItem [] { new TaskItem (frameworkDir) };
+			task.OnlyFilterFrameworks = true;
+
+			// Execute the task
+			ExecuteTask (task);
+
+			// Assert: Task should succeed and the framework should be processed correctly
+			Assert.That (task.FrameworkToPublish, Is.Not.Null);
+			Assert.That (task.FrameworkToPublish.Length, Is.EqualTo (1), $"Framework should be included for {platform}");
+		}
+
+		[Test]
+		[TestCase (ApplePlatform.iOS, "TestFramework.framework")]
+		[TestCase (ApplePlatform.TVOS, "TestFramework.framework")]
+		[TestCase (ApplePlatform.MacOSX, "TestFramework.framework")]
+		[TestCase (ApplePlatform.MacCatalyst, "TestFramework.framework")]
+		public void TestDefaultFrameworkExecutablePath (ApplePlatform platform, string frameworkName)
+		{
+			TestDefaultFrameworkExecutablePathForPlatform (platform, frameworkName);
+		}
+
+		void TestDefaultFrameworkExecutablePathForPlatform (ApplePlatform platform, string frameworkName)
+		{
+			var isDesktopPlatform = platform == ApplePlatform.MacOSX || platform == ApplePlatform.MacCatalyst;
+
+			// Arrange: Create a framework without Info.plist (or with default CFBundleExecutable)
+			var tempDir = Cache.CreateTemporaryDirectory ();
+			var frameworkDir = Path.Combine (tempDir, platform.AsString (), frameworkName);
+			Directory.CreateDirectory (frameworkDir);
+
+			if (isDesktopPlatform) {
+				// macOS and MacCatalyst structure
+				var versionsDir = Path.Combine (frameworkDir, "Versions", "A");
+				var resourcesDir = Path.Combine (versionsDir, "Resources");
+				Directory.CreateDirectory (resourcesDir);
+
+				// Create symlinks as they exist in real frameworks
+				Directory.CreateSymbolicLink (Path.Combine (frameworkDir, "Resources"), "Versions/A/Resources");
+			}
+
+			var expectedExecutable = Path.Combine (frameworkDir, "TestFramework");
+			File.WriteAllBytes (expectedExecutable, CreateMinimalMachODylib ());
+
+			// Act: Create and execute the task
+			var task = CreateTask<FilterStaticFrameworks> ();
+			task.FrameworkToPublish = new ITaskItem [] { new TaskItem (frameworkDir) };
+			task.OnlyFilterFrameworks = true;
+
+			// Execute the task
+			ExecuteTask (task);
+
+			// Assert: Task should succeed and use default framework executable path
+			Assert.That (task.FrameworkToPublish, Is.Not.Null);
+			Assert.That (task.FrameworkToPublish.Length, Is.EqualTo (1), $"Framework should be included for {platform}");
+		}
+
+		[Test]
+		public void TestNonFrameworkPath ()
+		{
+			// Arrange: Use a non-framework path
+			var tempDir = Cache.CreateTemporaryDirectory ();
+			var nonFrameworkPath = Path.Combine (tempDir, "regular_file.dylib");
+			File.WriteAllText (nonFrameworkPath, "mock dylib");
```

Comment:
Create a valid Mach-O file here by using the existing CreateMinimalMachODylib method.